src/java.base/unix/classes/java/io/UnixFileSystem.java

```diff
@@ -29,11 +29,18 @@
 import jdk.internal.util.StaticProperty;
 
 final class UnixFileSystem extends FileSystem {
-
     private final char slash;
     private final char colon;
     private final String userDir;
 
+    private String getPathForSysCalls(String path) {
+        return path.isEmpty() ? getCWD().getPath() : path;
```

Comment:
The Windows implementation has a guard for path == null. Is it superfluous there, or should it be added here?
```
    private String getPathForWin32Calls(String path) {
        return (path != null && path.isEmpty()) ? getCWD().getPath() : path;
    }
```

Reply:
Without the `null` check on Windows there were test failures which I do not see on Unix.

Reply:
OK - let's not poke the beast ;-)